packages/firebase_database/firebase_database/lib/src/query.dart

```diff
@@ -81,6 +81,30 @@ class Query {
   /// Listens for a single value event and then stops listening.
   Future<DataSnapshot> once() async => (await onValue.first).snapshot;
 
+  /// Gets the most up-to-date result for this query.
+  Future<DataSnapshot?> get() async {
+    final result = await _database._channel.invokeMethod<Map<dynamic, dynamic>>(
+      'Query#get',
+      <String, dynamic>{
+        'app': _database.app?.name,
+        'databaseURL': _database.databaseURL,
+        'path': path,
+      },
+    );
+    if (result == null || result is! Map) {
```

Comment:
`result is! Map` is never reached

I'm not a fan of this `if` + exception. Maybe we want `result!.containsKey('error')`?

Reply:
Seems like I can depend on result not being null. So I'll just check for the key. Thanks!

Reply:
I was wrong added your suggestion thanks!

Reply:
There is still an instance of `if` + exception though. I'm happy to rework to avoid if you think it is needed.